craft_application/util/platforms.py

```diff
@@ -36,6 +39,15 @@ def convert_architecture_deb_to_platform(arch: str) -> str:
     return _ARCH_TRANSLATIONS_DEB_TO_PLATFORM.get(arch, arch)
 
 
+@functools.lru_cache(maxsize=1)
+def get_host_base() -> bases.BaseName:
```

Comment:
Should `OsRelease` errors be caught here?  Or is the application responsible for this when creating a build plan?

Reply:
I thought about this but for now I think we should just let the `OsRelease` errors propagate up. The reasons for this are:

1) For now this function is actually only used by tests (as a kind of future-proofing them when we *do* restrict destructive runs to the matching bases);
2) Any environment where `OsRelease` would fail seems like it might have bigger issues - that is, I don't really know how to test failures here without the tests themselves being too artificial (and thus possibly counter-productive).